src/test/java/service/ReaderImplTest.java

```diff
@@ -0,0 +1,51 @@
+package service;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ReaderImplTest {
+    private static final String FIRST_INPUT_PATH = "src/test/resources/testInput.csv";
+    private static final String EMPTY_INPUT_PATH = "src/test/resources/emptyInput.csv";
+    private static final String NON_EXISTING_FILE_PATH = "src/test/resources/inputNONExisting.csv";
+    private static Reader reader;
+
+    @BeforeClass
+    public static void beforeClass() {
+        reader = (Reader) new ReaderImpl();
```

Comment:
```suggestion
        reader = new ReaderImpl();
```

why do we need cast here?